apps/aechannel/src/aesc_checks.erl

```diff
@@ -0,0 +1,100 @@
+%% @doc This modules provides stateless checks for various properties throughout
+%% the channels application.
+-module(aesc_checks).
+
+-export([ state_password/1
+        , state_password_in_opts/1
+        , known_role/1
+        , amounts/1
+        , account/2
+        , accounts/3
+        , lock_period/1
+        ]).
+
+-include("aesc_codec.hrl").
+-include("aechannel.hrl").
+
+%% TODO: Make this configurable
+%% No need for a stronger password policy
+%% This check is only here to ensure that someone doesn't enter a 1-2 character password
+-define(STATE_PASSWORD_MINIMUM_LENGTH, 6).
+
+%% ==================================================================
+%% Checks
+
+-spec state_password_in_opts(map()) -> ok | {error, password_required_since_lima | invalid_password}.
```

Comment:
I see that code is just moved around, this got my attention, will never return 'password_required_since_lima`